pkg/client/gen_test.go

```diff
@@ -156,34 +152,71 @@ func TestGenerateManifestSSH(t *testing.T) {
 		}, {
 			name: "Only e2e",
 			inputcm: &client.GenConfig{
-				E2EConfig: &client.E2EConfig{},
-				Config: fromConfig(func(c *config.Config) *config.Config {
-					c.PluginSelections = []plugin.Selection{plugin.Selection{Name: "e2e"}}
-					return c
-				}),
+				E2EConfig:      &client.E2EConfig{},
+				Config:         newConfigWithoutUUID(),
+				DynamicPlugins: []string{"e2e"},
 			},
 			goldenFile: filepath.Join("testdata", "e2e-default.golden"),
 		}, {
 			name: "Only systemd_logs",
 			inputcm: &client.GenConfig{
-				E2EConfig: &client.E2EConfig{},
-				Config: fromConfig(func(c *config.Config) *config.Config {
-					c.PluginSelections = []plugin.Selection{plugin.Selection{Name: "systemd-logs"}}
-					return c
-				}),
+				E2EConfig:      &client.E2EConfig{},
+				Config:         newConfigWithoutUUID(),
+				DynamicPlugins: []string{"systemd-logs"},
```

Comment:
I updated these tests but I need to think through the implications a bit more. What if someone does just send a plugin selection and no other plugins?

Do we want to treat it like they are saying to run nothing? Error?

Again, the aggregator should be getting the plugin info from the mounted configmaps not from the config itself. But I just need to consider what these tests were really about and how consumers may be affected.

Reply:
I think especially in this context, updating the tests are fine. The tests are about whether or not given an appropriate configuration whether or not we generate the yaml we expect.

I do, however, need to define and test the behavior if someone wants to ignore these changes. If they just send the `Plugin` without setting either of the new fields. Also need to consider interaction with #684. TBD